ui/app/appointments/controllers/manage/list/appointmentsListViewController.js

```diff
@@ -232,13 +233,37 @@ angular.module('bahmni.appointments')
             };
 
             $scope.isValidAction = function (action) {
-                if (!$scope.selectedAppointment) {
+                if (!$scope.isUserAllowedToPerform() || !$scope.selectedAppointment) {
                     return false;
                 }
                 var status = $scope.selectedAppointment.status;
                 var allowedActions = $scope.allowedActionsByStatus.hasOwnProperty(status) ? $scope.allowedActionsByStatus[status] : [];
                 return _.includes(allowedActions, action);
             };
 
+            var isCurrentUserHavePrivilege = function (privilege) {
+                return !_.isUndefined(_.find($rootScope.currentUser.privileges, function (userPrivilege) {
+                    return userPrivilege.name === privilege;
+                }));
+            };
+
+            $scope.isUserAllowedToPerform = function () {
+                if (isCurrentUserHavePrivilege($scope.manageAppointmentPrivilege)) {
+                    return true;
+                }
+                else if (isCurrentUserHavePrivilege($scope.ownAppointmentPrivilege) && $scope.selectedAppointment) {
+                    return _.isEmpty($scope.selectedAppointment.providers) ||
+                        !_.isUndefined(_.find($scope.selectedAppointment.providers, function (provider) {
```

Comment:
@sai789, can you please pull this block into a different method so that its readable